test/operators/second_quantization/test_spin_op.py

```diff
@@ -24,8 +25,20 @@
 
 from qiskit_nature.operators import SpinOp
```

Comment:
Could you please also add some tests for `SpinOp.reduce()`? In fact, I think it may not do everything as expected in some cases. For example:
```python
In [1]: from qiskit_nature.operators.second_quantization import SpinOp

In [2]: op = SpinOp([('+_0 -_0', 1)], register_length=4)

In [3]: op
Out[3]: SpinOp([('X_0^2 I_1 I_2 I_3', (1+0j)), ('Y_0^2 I_1 I_2 I_3', (1+0j)), ('Z_0 I_1 I_2 I_3', (1+0j))], spin=1/2, register_length=4)

In [4]: op.reduce()
Out[4]: SpinOp([('Z_0 Y_1^2 X_2^2 I_3', (1+0j)), ('I_0 I_1 I_2 I_3', (1+0j)), ('I_0 I_1 I_2 I_3', (1+0j))], spin=1/2, register_length=4)

In [5]: op.reduce().reduce()
Out[5]: SpinOp([('I_0 Z_1 I_2 Y_3^2', (2+0j)), ('I_0 X_1^2 I_2 I_3', (1+0j))], spin=1/2, register_length=4)
```

Reply:
Thank you so much for the finding the bug.